go/vt/srvtopo/resolver.go

```diff
@@ -382,71 +194,144 @@ func (r *Resolver) ResolveEntityIds(ctx context.Context, keyspace string, entity
 			}
 			_, qs, err := r.stats.GetAggregateStats(target)
 			if err != nil {
-				return nil, nil, resolverError(err, target)
+				return resolverError(err, target)
 			}
 
 			// FIXME(alainjobart) we ignore the stats for now.
 			// Later we can fallback to another cell if needed.
 			// We would then need to read the SrvKeyspace there too.
 			target.Cell = ""
-			i = len(result)
+			s = len(result)
 			result = append(result, &ResolvedShard{
 				Target:       target,
 				QueryService: qs,
 			})
-			values = append(values, nil)
-			resolved[shard] = i
+			if ids != nil {
+				values = append(values, nil)
+			}
+			resolved[shard] = s
+		}
+		if ids != nil {
```

Comment:
It looks to me like this should be _inside_ the `if !ok` block, no?

Reply:
I don't think so, the logic is:
* if we haven't see that shard yet, create it. Otherwise, just use it.
* add the ID to the ids list for the shard

Reply:
Oh of course... len(values) != len(shards). Duh.